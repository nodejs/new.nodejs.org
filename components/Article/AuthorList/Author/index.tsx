import { useState } from 'react';
import { useIntl } from 'react-intl';
import Image from 'next/image';
import styles from './index.module.scss';
import type { FC } from 'react';

type AuthorProps = { username: string; size?: number };

const Author: FC<AuthorProps> = ({ username, size }) => {
  const githubUserName = username.trim();
  const githubLink = `https://github.com/${githubUserName}`;
  const githubImgLink = `https://github.com/${githubUserName}.png?size=${size}`;

  const intl = useIntl();

  const [authorImg, setAuthorImg] = useState(githubImgLink);

  const translation = intl.formatMessage(
    { id: 'components.article.author.githubLinkLabel' },
    { username }
  );

  return (
    <a
      className={styles.link}
      href={githubLink}
      aria-label={translation}
      key={username}
      target="_blank"
      rel="noopener noreferrer"
    >
      <Image
        alt={githubUserName}
        src={authorImg}
        placeholder="blur"
        blurDataURL="/placeholder-img.png"
        width={size}
        height={size}
        onError={() => setAuthorImg('/placeholder-img.png')}
      />
    </a>
  );
};

export default Author;
