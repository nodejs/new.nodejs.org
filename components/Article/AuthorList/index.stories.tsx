import AuthorList from './index';
import type { Meta as MetaObj, StoryObj } from '@storybook/react';

type Story = StoryObj<typeof AuthorList>;
type Meta = MetaObj<typeof AuthorList>;

export const Default: Story = {
  args: {
    authors: ['flaviocopes', 'MarkPieszak', 'mcollina', 'unavailable-author'],
  },
};

export default { component: AuthorList } as Meta;
