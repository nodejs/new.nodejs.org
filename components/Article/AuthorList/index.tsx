import { FormattedMessage } from 'react-intl';
import Author from './Author';
import styles from './index.module.scss';
import type { FC } from 'react';

type AuthorListProps = { authors: string[] };

const AuthorList: FC<AuthorListProps> = ({ authors }) => {
  if (authors.length) {
    return (
      <div className={styles.authorList}>
        <FormattedMessage id="components.article.authorList.title" />
        <ul>
          {authors.map(author => (
            <li key={author}>
              <Author username={author} key={author} size={60} />
            </li>
          ))}
        </ul>
      </div>
    );
  }

  return null;
};

export default AuthorList;
