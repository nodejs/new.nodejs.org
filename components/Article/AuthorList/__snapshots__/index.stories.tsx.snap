// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article/AuthorList Default smoke-test 1`] = `
<div class="AuthorList_authorList__YPp0H">
  Article Authors
  <ul>
    <li>
      <a class="Author_link__6oI83"
         href="https://github.com/flaviocopes"
         aria-label="flaviocopes Github - opens in new tab"
         target="_blank"
         rel="noopener noreferrer"
      >
        <img alt="flaviocopes"
             loading="lazy"
             width="60"
             height="60"
             decoding="async"
             data-nimg="1"
             srcset="https://github.com/flaviocopes.png?size=60?w=64&amp;q=75 1x, https://github.com/flaviocopes.png?size=60?w=128&amp;q=75 2x"
             src="https://github.com/flaviocopes.png?size=60?w=128&amp;q=75"
             style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat; background-image: url(&quot;/placeholder-img.png&quot;);"
        >
      </a>
    </li>
    <li>
      <a class="Author_link__6oI83"
         href="https://github.com/MarkPieszak"
         aria-label="MarkPieszak Github - opens in new tab"
         target="_blank"
         rel="noopener noreferrer"
      >
        <img alt="MarkPieszak"
             loading="lazy"
             width="60"
             height="60"
             decoding="async"
             data-nimg="1"
             srcset="https://github.com/MarkPieszak.png?size=60?w=64&amp;q=75 1x, https://github.com/MarkPieszak.png?size=60?w=128&amp;q=75 2x"
             src="https://github.com/MarkPieszak.png?size=60?w=128&amp;q=75"
             style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat; background-image: url(&quot;/placeholder-img.png&quot;);"
        >
      </a>
    </li>
    <li>
      <a class="Author_link__6oI83"
         href="https://github.com/mcollina"
         aria-label="mcollina Github - opens in new tab"
         target="_blank"
         rel="noopener noreferrer"
      >
        <img alt="mcollina"
             loading="lazy"
             width="60"
             height="60"
             decoding="async"
             data-nimg="1"
             srcset="https://github.com/mcollina.png?size=60?w=64&amp;q=75 1x, https://github.com/mcollina.png?size=60?w=128&amp;q=75 2x"
             src="https://github.com/mcollina.png?size=60?w=128&amp;q=75"
             style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat; background-image: url(&quot;/placeholder-img.png&quot;);"
        >
      </a>
    </li>
    <li>
      <a class="Author_link__6oI83"
         href="https://github.com/unavailable-author"
         aria-label="unavailable-author Github - opens in new tab"
         target="_blank"
         rel="noopener noreferrer"
      >
        <img alt="unavailable-author"
             loading="lazy"
             width="60"
             height="60"
             decoding="async"
             data-nimg="1"
             srcset="https://github.com/unavailable-author.png?size=60?w=64&amp;q=75 1x, https://github.com/unavailable-author.png?size=60?w=128&amp;q=75 2x"
             src="https://github.com/unavailable-author.png?size=60?w=128&amp;q=75"
             style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat; background-image: url(&quot;/placeholder-img.png&quot;);"
        >
      </a>
    </li>
  </ul>
</div>
`;
